src/adapter/debugAdapter.ts

```diff
@@ -260,6 +262,38 @@ export class DebugAdapter {
     return {};
   }
 
+  async _toggleSkipFileStatus(params: Dap.ToggleSkipFileStatusParams): Promise<Dap.ToggleSkipFileStatusResult> {
+    if (params.resource) {
+      params.resource = utils.fileUrlToPath(params.resource);
+    }
+
+    const dapSource = await this._isSourceInCurrentStack(params);
+    if (dapSource && this.sourceContainer.scriptSkipper) {
+      // TODO@Shennie set up script toggling regardless of whether skipfiles was passed from the beginning
+      const source = this.sourceContainer.source(dapSource);
+      if (source) {
+        source._blackboxed = !source._blackboxed;
+      }
+      this.sourceContainer.scriptSkipper.toggleSkipFileStatus(params.resource);
+      this._withThread(thread => thread.pause());
+    }
+    return {};
+  }
+
+  private async _isSourceInCurrentStack(params: Dap.ToggleSkipFileStatusParams): Promise<Dap.Source | undefined> {
+    const currentStack = await this._onStackTrace({ threadId: this._thread?.id! });
+    const stackFrames = (currentStack as Dap.StackTraceResult).stackFrames;
+
+    let foundStackFrame;
+    if (params.resource && stackFrames) {
+      foundStackFrame = stackFrames.find(frame => frame.source && frame.source.path?.toLocaleLowerCase() === params.resource?.toLocaleLowerCase());
+    } else {
+      foundStackFrame = stackFrames.find(frame => frame.source && frame.source.sourceReference === params.sourceReference);
+    }
```

Comment:
Do you need any special handling for dealing with sourcemapped scripts here?

Reply:
I... don't think so? When I tested this the stackFrame's source path is the correct source value for sourcemapped scripts. As in, when it's running index.js, the stackFrame has the index.ts as its source. But there could definitely be some cases that I'm missing . 

@roblourens Are there special cases for handling sourcemapped scripts?

Reply:
Where does it get mapped back to `index.js` though? It looks like if this is `index.ts`, then that goes directly to the ScriptSkipper and would get sent to the runtime, when it needs to be resolved to `index.js` because node doesn't know what `index.ts` is.